src/JuMP.jl

```diff
@@ -441,6 +441,63 @@ function getdual(v::Variable)
     return v.m.redCosts[v.col]
 end
 
+# Returns the variable dual, throwing an error if it is not available
+function getvardualaux(m::Model)
+    if !m.internalModelLoaded
+        error("Model not solved.")
+    end
+    vardual = MathProgBase.getvardual(m.internalModel)
+    if length(vardual) != m.numCols
+        error("Dual solution not available. Check that the model was properly solved and no integer variables are present.")
+    end
+    vardual
+end
+
+
+function getdual(arr::Matrix{Variable})
+    ret = similar(arr, Float64)
+    # return immediately for empty array
+    if isempty(ret)
+        return ret
+    end
+    m = first(arr).m
+    if size(arr, 1) != size(arr, 2)
+        sdpvar = false
+    else
+        sdpvar = true
+        n = size(arr, 1)
+        start = arr[1,1].col
+        cur = start
+        for i in 1:n
+            for j in i:n
+                if arr[i,j].m != m
+                    sdpvar = false
+                end
+                if arr[i,j].col != cur
+                    sdpvar = false
+                end
+                cur += 1
+                if arr[i,j] != arr[j,i]
+                    sdpvar = false
+                end
+            end
+        end
+        if sdpvar && !((:SDP, start:arr[n,n].col) in m.varCones)
+            sdpvar = false
+        end
+    end
+    if !sdpvar
+        error("getdual on a matrix that is not an SDP variable is not supported")
```

Comment:
Then it seems that my approach is completely wrong.
In fact it already worked without my new function :-P